test/parallel/test-stream-writable-needdrain-state.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+require('../common');
+const stream = require('stream');
+const assert = require('assert');
+
+const transform = new stream.Transform({
+  transform: _transform,
+  highWaterMark: 1
+});
+
+function _transform(chunk, encoding, cb) {
+  cb();
+}
+
+assert.strictEqual(transform._writableState.needDrain, false);
+
+transform.write('asdasd', () => {
```

Comment:
this callback should be wrapped in a `common.mustCall()` to ensure that it's actually called.